### PR TITLE
Modify discounts listing in Customer page to use Grid - Delete modal

### PR DESCRIPTION
**Pull request by sowbiba**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Modify discounts listing in Customer page to use Grid instead of table. This enables use to add a confirmation modal when deleting a row from a grid.
Customers > View > Discounts.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Partially Fixes #17847
| How to test?  | Go to Customers > View > Discounts in the BO, Try to delete a row, you will have a bootstrap modal to confirm deletion

This may have impact on automated tests @PrestaShop/qa-automation

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20799)

### CODE REVIEW

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Actions/Row/delete_customer_discount.html.twig [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+{% set class = 'btn tooltip-link js-submit-row-action' %}
+
+{% if attributes.class is defined %}
+  {% set class = class ~ ' ' ~ attributes.class %}
+{% endif %}
+
+{% set confirmation_message = action.options.confirm_message %}
+
+<a class="{{ class }} grid-{{ action.name|lower }}-row-link"
+   href="#"
```
Comment (NeOMakinG) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Actions/Row/delete_customer_discount.html.twig]: If it's not a link, prefer using a button for any other actions

**Review by NeOMakinG — CHANGES_REQUESTED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Actions/Row/delete_customer_discount.html.twig (hunk 1 above)
Comment (sowbiba) [reply src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Actions/Row/delete_customer_discount.html.twig]: fixed

File: src/Core/Grid/Action/Row/Type/Customer/DeleteCustomerDiscountRowAction.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Action\Row\Type\Customer;
```
Comment (Progi1984) [inline src/Core/Grid/Action/Row/Type/Customer/DeleteCustomerDiscountRowAction.php]:
```suggestion
 */
 
 declare(strict_types=1);

namespace PrestaShop\PrestaShop\Core\Grid\Action\Row\Type\Customer;
```

File: src/Core/Grid/Action/Row/Type/Customer/DeleteCustomerDiscountRowAction.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+ 
+ declare(strict_types=1);
```
Comment (Progi1984) [inline src/Core/Grid/Action/Row/Type/Customer/DeleteCustomerDiscountRowAction.php]:
```suggestion
declare(strict_types=1);
```

File: src/Core/Grid/Action/Row/Type/Customer/EditCustomerDiscountRowAction.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Action\Row\Type\Customer;
```
Comment (Progi1984) [inline src/Core/Grid/Action/Row/Type/Customer/EditCustomerDiscountRowAction.php]:
```suggestion
 */

declare(strict_types=1);

namespace PrestaShop\PrestaShop\Core\Grid\Action\Row\Type\Customer;
```

File: src/Core/Grid/Data/Factory/CustomerDiscountGridDataFactory.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
```
Comment (Progi1984) [inline src/Core/Grid/Data/Factory/CustomerDiscountGridDataFactory.php]:
```suggestion
 */

declare(strict_types=1);

namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
```

Comment (sowbiba) [reply src/Core/Grid/Data/Factory/CustomerDiscountGridDataFactory.php]: fixed

File: src/Core/Grid/Action/Row/Type/Customer/EditCustomerDiscountRowAction.php (hunk 4 above)
Comment (sowbiba) [reply src/Core/Grid/Action/Row/Type/Customer/EditCustomerDiscountRowAction.php]: fixed

**Review by Progi1984 — DISMISSED** [review]

File: src/Core/Grid/Action/Row/Type/Customer/DeleteCustomerDiscountRowAction.php (hunk 3 above)
Comment (sowbiba) [reply src/Core/Grid/Action/Row/Type/Customer/DeleteCustomerDiscountRowAction.php]: fixed

**Review by NeOMakinG — DISMISSED** [review]
lgtm for the front

File: src/Core/Grid/Data/Factory/CustomerDiscountGridDataFactory.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        $this->contextLangId = $contextLangId;
+        $this->customer = $customer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $discounts = \CartRule::getCustomerCartRules(
```
Comment (PierreRambaud) [inline src/Core/Grid/Data/Factory/CustomerDiscountGridDataFactory.php]:
```suggestion
        $discounts = CartRule::getCustomerCartRules(
```

**Review by PierreRambaud — APPROVED** [review]

**Comment by boubkerbribri** [comment]
> This may have impact on automated tests @PrestaShop/qa-automation

@sowbiba Thanks for the ping, There is no impact on UI tests.

**Review by Progi1984 — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @sowbiba,

I tried to delete a voucher but nothing happens.
image

I attached a screen record
https://drive.google.com/file/d/1IZjhxAIDC0wMm189mDKVXU6bsAeAtP4O/view?usp=sharing

Thanks!

**Comment by sowbiba** [comment]
Hi @khouloudbelguith 
Can you check your env when you executed the test please ?
https://drive.google.com/file/d/1XTlTi-PMymEMynR67pb8srdpBD6vnZVd/view

**Comment by khouloudbelguith** [comment]
Hi @sowbiba,

I just make a rebuilt assets and it is ok :heavy_check_mark: 
image
https://drive.google.com/file/d/1fpWIxPToc2FXjNhtnb2bqjr6fFMhU56s/view?usp=sharing
Thanks!

**Comment by Progi1984** [comment]
Thanks @sowbiba & @khouloudbelguith

**Pull request merged**